### DISCUSSION TITLE
setting OPTIX_FORCE_DEPRECATED_LAUNCHER flag

### DISCUSSION THREAD

**Discussion by MDurgaKeerthi** [owl-project/owl#199] [Q&A]
Recently, there was an issue with printf in optix kernel and the fix was to set OPTIX_FORCE_DEPRECATED_LAUNCHER environment variable to 1/ON. I added this command to cmake file but it doesn't work yet. Could you please help me with figuring out where to set this variable?

https://forums.developer.nvidia.com/t/printf-not-working-in-optix-kernel/246586/2

**Comment by MDurgaKeerthi** [comment]
The fix is to do `export OPTIX_FORCE_DEPRECATED_LAUNCHER=1` and not add it to the cmake file.

**Reply by ingowald** [reply]
huh, intersting. i wasn't even aware of that issue - i use printf all the time, but never ran into that. good to know there's a workaround, though. I assume that's since been fixed in newer drivers?

**Reply by MDurgaKeerthi** [reply]
afaik, its fixed for rtx 30 series and newer gpus. We are still working on 20 series tho
